presto-main/src/main/java/com/facebook/presto/execution/KillQueryTask.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution;
+
+import java.util.Map;
+import java.util.Map.Entry;
+
+import io.airlift.http.client.FullJsonResponseHandler.JsonResponse;
+import io.airlift.http.client.FullJsonResponseHandler;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.http.client.Request;
+import io.airlift.http.client.jetty.JettyHttpClient;
+import io.airlift.json.JsonCodec;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.client.ClientSession;
+import com.facebook.presto.client.PrestoHeaders;
+import com.facebook.presto.client.QueryResults;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.server.PrestoServer;
+import com.facebook.presto.sql.tree.KillQuery;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.net.HttpHeaders.USER_AGENT;
+import static io.airlift.http.client.FullJsonResponseHandler.createFullJsonResponseHandler;
+import static io.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static io.airlift.http.client.Request.Builder.prepareDelete;
+import static io.airlift.http.client.StaticBodyGenerator.createStaticBodyGenerator;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class KillQueryTask
+        implements DataDefinitionTask<KillQuery>
+{
+    private static final String USER_AGENT_VALUE = KillQueryTask.class.getSimpleName() +
+            "/" +
+            firstNonNull(KillQueryTask.class.getPackage().getImplementationVersion(), "unknown");
+
+    @Override
+    public String getName()
+    {
+        return "KILL QUERY";
+    }
+
+    @Override
+    public void execute(KillQuery statement, Session session, Metadata metadata, QueryStateMachine stateMachine)
+    {
+        ClientSession clientSession = session.toClientSession(PrestoServer.getDiscoveryUri(), false);
```

Comment:
Ah, yes, that creates a circular dependency because `SqlQueryManager` requires `QueryExecutionFactory`. I'll get back to you with a solution.